### ISSUE TITLE
Add Mission commands in Mission constructor

### ISSUE THREAD

**Issue by docprofsky** [TeamCautionRobotics/2017-FRC-robot#32]
We should be able to add commands to a Mission in its constructor.

The constructor would be called like this.
```Java
new Mission(id,
    commandFactory.alignWithVision(),
    commandFactory.moveStraight(false, 0.4, 0.5),
    commandFactory.delay(0.3));
```

**Comment by docprofsky** [comment]
Also add a `enableControls` field.
And change `int id` to `String name` for the missions.

**Comment by TaeSeanDo** [comment]
How do I avoid limiting the number of commands?

**Comment by TaeSeanDo** [comment]
never mind

**Comment by TaeSeanDo** [comment]
Can I use Lists instead of ArrayLists?

**Comment by docprofsky** [comment]
No, for two reasons:

1. Java's arrays are fixed size, so other things (mission script parsing) become more complicated.
2. Java's `ArrayList` is the most similar to Python lists, which I believe is what you want. Also, Java does not have a plain `List` object.

**Comment by docprofsky** [comment]
This is closed by #35.
